### PR TITLE
Feature/360/91 cosmosdb emulator

### PR DESCRIPTION
**Pull request by algattik**
## What this PR changes/adds:

Use Cosmos DB Emulator in CI. The integration test pipeline currently runs only in upstream main, as it requires access to an Azure Cosmos DB instance.

## Why it does that

This allows fully validating changes that affect the Cosmos DB related extensions *before* merging to upstream main, rather than *after*.

## Further notes

One test had a logical issue due to incomplete data cleanup, and was fixed.

Pipeline was run 100 times without failures.

Use gradle-build-action as it does a better job of caching than setup-java, to accelerate the pipeline in most runs (from 12 min without cache to 6 min)

## Linked Issue(s)

#91

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [x] documented public classes/methods?
- [x] added/updated relevant documentation?

### CODE REVIEW

File: extensions/azure/azure-test/src/testFixtures/java/org/eclipse/dataspaceconnector/azure/testfixtures/CosmosTestClient.java [hunk 1]
```diff
@@ -27,7 +28,7 @@
 
     static CosmosClient createClient() {
         var cosmosKey = propOrEnv("COSMOS_KEY", null);
```
Comment (cpeeyush) [inline extensions/azure/azure-test/src/testFixtures/java/org/eclipse/dataspaceconnector/azure/testfixtures/CosmosTestClient.java:30]: from where COSMOS_KEY values comes now as above you removed it from the pipeline environment variable ?

File: extensions/azure/azure-test/src/testFixtures/java/org/eclipse/dataspaceconnector/azure/testfixtures/CosmosTestClient.java [hunk 2]
```diff
@@ -27,7 +28,7 @@
...
-        if (cosmosKey != null) {
+        if (!StringUtils.isNullOrBlank(cosmosKey)) {
```
Comment (cpeeyush) [inline extensions/azure/azure-test/src/testFixtures/java/org/eclipse/dataspaceconnector/azure/testfixtures/CosmosTestClient.java:31]: is this changes needed here or part of a general improvement ? or do you expect key can be Blank now ?

File: .github/workflows/integrationtests.yaml [hunk 3]
```diff
@@ -51,7 +45,14 @@ jobs:
...
           distribution: 'temurin'
           java-version: '11'
-          cache: 'gradle'
+
+      - name: Setup Gradle cache
+        uses: gradle/gradle-build-action@v2
+
+      - name: Launch Cosmos DB Emulator
+        run: |
+          Import-Module "$env:ProgramFiles\Azure Cosmos DB Emulator\PSModules\Microsoft.Azure.CosmosDB.Emulator"
```
Comment (Izzzu) [inline .github/workflows/integrationtests.yaml:54]: out of curiosity: does this emulator comes together with windows-2019? If not how is it installed?

**Review by Izzzu — APPROVED** [review]

File: extensions/azure/cosmos/contract-definition-store-cosmos/src/test/java/org/eclipse/dataspaceconnector/contract/definition/store/CosmosContractDefinitionStoreIntegrationTest.java [hunk 4]
```diff
@@ -47,6 +48,8 @@
     private TypeManager typeManager;
     private CosmosContractDefinitionStore store;
 
+    private static final String PARTITION_KEY_NEW = "test-ap-id1-new";
```
Comment (cpeeyush) [inline extensions/azure/cosmos/contract-definition-store-cosmos/src/test/java/org/eclipse/dataspaceconnector/contract/definition/store/CosmosContractDefinitionStoreIntegrationTest.java:51]: nit: may be name of the variable could be improved. Understand it is because of the existing values.

**Review by cpeeyush — APPROVED** [review]

File: extensions/azure/azure-test/src/testFixtures/java/org/eclipse/dataspaceconnector/azure/testfixtures/CosmosTestClient.java (hunk 2 above)
Comment (algattik) [reply extensions/azure/azure-test/src/testFixtures/java/org/eclipse/dataspaceconnector/azure/testfixtures/CosmosTestClient.java:31]: It wouldn't be anymore given pipeline changes, but could be if we add other test environments, so it seemed reasonable to leave the change as an improvement

File: .github/workflows/integrationtests.yaml (hunk 3 above)
Comment (algattik) [reply .github/workflows/integrationtests.yaml:54]: Added comment

File: extensions/azure/azure-test/src/testFixtures/java/org/eclipse/dataspaceconnector/azure/testfixtures/CosmosTestClient.java (hunk 1 above)
Comment (algattik) [reply extensions/azure/azure-test/src/testFixtures/java/org/eclipse/dataspaceconnector/azure/testfixtures/CosmosTestClient.java:30]: Added doc in markdown
